src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -170,10 +175,24 @@ func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, erro
 		return nil, err
 	}
 
-	return mod.aiCache.getResponse(numaNode)
+	cacheResp, err := mod.aiCache.getResponse(numaNode)
+	if err != nil {
+		return nil, err
+	}
+
+	req := NewProcMonRequest(pid, drpc.MethodGetAttachInfo.ID())
+	mod.monitor.request <- req
+
+	return cacheResp, err
 }
 
+// handleDisconnect crafts a new request for the process monitor to inform the
+// monitor that a process is exiting. Even though the process is terminating
+// cleanly disconnect will inform the control plane of any outstanding handles
+// that the process held open.
 func (mod *mgmtModule) handleDisconnect(reqb []byte, pid int32) ([]byte, error) {
```

Comment:
If the final implementation returns nil, nil, wouldn't it be cleaner to rework the function prototype to something else?  Can something generate an error that you are planning to return in a future version?

Reply:
Fixed both of these.